lib/data-types.js

```diff
@@ -789,14 +790,38 @@ UUID.prototype.validate = function validate(value, options) {
   return true;
 };
 
+/**
+ * A column storing a modified uuidv1 as binary(16).
+ * @see https://www.percona.com/blog/2014/12/19/store-uuid-optimized-way/
+ *
+ * @namespace DataTypes.BINARYUUID
+ */
+function BINARYUUID() {
+  if (!(this instanceof BINARYUUID)) return new BINARYUUID();
+}
+inherits(BINARYUUID, ABSTRACT);
+
+BINARYUUID.prototype.key = UUID.key = 'BINARY(16)';
+BINARYUUID.prototype.escape = false;
+
+BINARYUUID.prototype.validate = function validate(value) {
+  if (!Buffer.isBuffer(value) || value.length !== 16) {
+    throw new sequelizeErrors.ValidationError(util.format('%j is not a valid binary uuid (buffer)', value));
+  }
+  return true;
+};
+
 /**
  * A default unique universal identifier generated following the UUID v1 standard
  *
+ * @param {boolean} [binary=false] Whether UUID should be transformed to binary(16)
  * @namespace DataTypes.UUIDV1
  */
-function UUIDV1() {
-  if (!(this instanceof UUIDV1)) return new UUIDV1();
+function UUIDV1(binary) {
```

Comment:
I suggest you make this parameter an object instead, avoid future breaking changes if we wish to add more optional parameters.